samza-core/src/main/java/org/apache/samza/operators/impl/OperatorImpl.java

```diff
@@ -495,7 +498,9 @@ protected String getOpImplId() {
   }
 
   private HighResolutionClock createHighResClock(Config config) {
-    if (new MetricsConfig(config).getMetricsTimerEnabled()) {
+    MetricsConfig metricsConfig = new MetricsConfig(config);
+    // The timer metrics calculation here is only enabled for debugging
+    if (metricsConfig.getMetricsTimerDebugEnabled()) {
```

Comment:
doesn't this have to be both Timer and TimerDebug enabled?

Reply:
My old impl has both, but i was thinking it's clearer to have them separated so each flag controls a different set of timers. If you think the first one should have an overall control, I can add that back.